include/deal.II/base/table.h

```diff
@@ -2661,60 +2893,76 @@ namespace TransposeTableIterators
 
 
 
-  template <typename T, bool Constness>
-  inline const T &
-  AccessorBase<T, Constness>::value() const
+  template <typename TableType, bool Constness, Storage storage_order>
+  inline const typename AccessorBase<TableType, Constness, storage_order>::
+    value_type &
+    AccessorBase<TableType, Constness, storage_order>::value() const
   {
     assert_valid_linear_index();
     return this->container->values[linear_index];
   }
 
 
 
-  template <typename T, bool Constness>
-  inline typename AccessorBase<T, Constness>::size_type
-  AccessorBase<T, Constness>::AccessorBase::row() const
+  template <typename TableType, bool Constness, Storage storage_order>
+  inline typename AccessorBase<TableType, Constness, storage_order>::size_type
+  AccessorBase<TableType, Constness, storage_order>::AccessorBase::row() const
```

Comment:
I know this is pre-existing, but is there an `AccessorBase::` too many here?

Reply:
I am a bit frightened that this ever worked.

Reply:
The name of the class is available inside the class. I'm not terribly surprised that it works, it just looks funny :-)